app/src/main/java/net/gsantner/markor/format/asciidoc/AsciidocSyntaxHighlighter.java

```diff
@@ -0,0 +1,324 @@
+    // black on yellow
+    private static final int AD_FORECOLOR_HIGHLIGHT = Color.BLACK;
+
+    // ..._COLOR_LIGHT_... = Color for light theme
+    // TODO: too little contrast to normal font with the TOL_ colors
+//    private static final int AD_COLOR_LIGHT_COMMENT = TOL_DARK_GRAY;
+//    private static final int AD_COLOR_DARK_COMMENT = TOL_PALE_GRAY;
+//    // https://docs.oracle.com/javase/9/docs/api/javafx/scene/paint/Color.html
+//    // use Color.GRAY instead of TOL_GRAY, because Color.GRAY is the middle between white and black: #808080
+//    // alternative: TOL_PALE_GRAY vs. TOL_DARK_GRAY
+//    private static final int AD_COLOR_COMMENT = Color.GRAY; //TOL_GRAY;
+    private static final int AD_FORECOLOR_LIGHT_COMMENT = Color.GRAY;
+    private static final int AD_FORECOLOR_DARK_COMMENT = Color.GRAY;
+
+
+    private boolean _highlightLineEnding;
+    private boolean _highlightCodeChangeFont;
+    private boolean _highlightBiggerHeadings;
+    private boolean _highlightCodeBlock;
+
+
+    public AsciidocSyntaxHighlighter(AppSettings as) {
+        super(as);
+    }
+
+    @Override
+    public SyntaxHighlighterBase configure(Paint paint) {
+        _highlightLineEnding = _appSettings.isAsciidocHighlightLineEnding();
+// TODO: does not work yet
+        _highlightBiggerHeadings = _appSettings.isAsciidocBiggerHeadings();
+        _highlightCodeChangeFont = _appSettings.isHighlightCodeMonospaceFont();
+        _highlightCodeBlock = _appSettings.isHighlightCodeBlock();
+        _delay = _appSettings.getAsciidocHighlightingDelay();
+        return super.configure(paint);
+    }
+
+    @Override
+    protected void generateSpans() {
+        createTabSpans(_tabSize);
+        // TODO: understand: what happens here?
+        createUnderlineHexColorsSpans();
+        // TODO: createSmallBlueLinkSpans() - font is very small, currently general setting: 85% of common size
+        // TODO: implement some kind of AsciiDoc link later
+        // hard to read on dark theme, but this is a general question for all formats,
+        // not AsciiDoc specific
+        // also it uses private static String formatLink(String text, String link), which is
+        // adapted for Markdown
+        // maybe, needs to be adapted for AsciiDoc?
+        // but not in the current Pull Request
+        // createSmallBlueLinkSpans();
+
+        //TODO: doesn't yet work, but it is called
+        if (_highlightBiggerHeadings) {
+            createSpanForMatches(HEADING_SIMPLE, new WrMarkdownHeaderSpanCreator(_spannable, _isDarkMode ? AD_FORECOLOR_DARK_HEADING : AD_FORECOLOR_LIGHT_HEADING, _textSize));
+        } else {
+//          createColorSpanForMatches(HEADING_SIMPLE, _isDarkMode ? AD_COLOR_DARK_HEADING : AD_COLOR_LIGHT_HEADING);
+            createSpanForMatches(HEADING_SIMPLE, new HighlightSpan().setForeColor(_isDarkMode ? AD_FORECOLOR_DARK_HEADING : AD_FORECOLOR_LIGHT_HEADING));
+        }
```

Comment:
what does not work? calling does nothing effective here? -> then remove it

Reply:
I want to understand, what should happen there and how. But I don't get it. If it just will be removed, then it could be forgotten.

Reply:
type `## Hello` in Markdown file with and without bigger headings enabled, and see yourself.

Reply:
In Markdown format it works. But I don't understand, what I should do to get it work in AsciiDoc.

Reply:
You apply highlighting (like text color) to headings in AsciiDoc right? Just in the same way you can also change the size.

You can check the Zim or Markdown Highlighter for reference.  
Or strike it and remove the todo/comment/condition if it's not viable